### PR TITLE
docs: rewrite event binding section and add example

### PR DESCRIPTION
**Pull request by kapunahelewong**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?
Rewrite of the Event Binding section of Template Syntax and create an example for just that section.

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information
This is the 5th installment of the Template Syntax rewrite. When Template Syntax is all finished, it can be broken into small docs, each with its own example. Examples are all item themed rather than hero themed.

### CODE REVIEW

File: aio/content/guide/template-syntax.md [hunk 1]
```diff
@@ -978,71 +965,78 @@ in response to the event, such as storing a value from the HTML control
...
 The component defines a `deleteRequest` property that returns an `EventEmitter`.
 When the user clicks *delete*, the component invokes the `delete()` method,
-telling the `EventEmitter` to emit a `Hero` object.
+telling the `EventEmitter` to emit an `Item` object.
 
-Now imagine a hosting parent component that binds to the `HeroDetailComponent`'s `deleteRequest` event.
+Now imagine a hosting parent component that binds to the `deleteRequest` event 
+of the `ItemDetailComponent`.
 
 <code-example path="template-syntax/src/app/app.component.html" linenums="false" title="src/app/app.component.html (event-binding-to-component)" region="event-binding-to-component">
```
Comment (brandonroberts) [inline aio/content/guide/template-syntax.md]: This example still refers to a `app-hero-detail` component.

File: aio/content/guide/template-syntax.md [hunk 2]
```diff
@@ -918,56 +917,44 @@ Note that a _style property_ name can be written in either
...
-A **name between parentheses** &mdash; for example, `(click)` &mdash;
-identifies the target event. In the following example, the target is the button's click event.
+As above, the target is the button's click event. 
 
-<code-example path="template-syntax/src/app/app.component.html" region="event-binding-1" title="src/app/app.component.html" linenums="false">
+<code-example path="event-binding/src/app/app.component.html" region="event-binding-1" title="src/app/app.component.html" linenums="false">
 </code-example>
 
-Some people prefer the `on-` prefix alternative, known as the **canonical form**:
+Alternatively, you can use the `on-` prefix, known as the canonical form:
```
Comment (brandonroberts) [inline aio/content/guide/template-syntax.md]: Remove "you can"

File: aio/content/guide/template-syntax.md [hunk 3]
```diff
@@ -978,71 +965,78 @@ in response to the event, such as storing a value from the HTML control
...
+The target event determines the shape of the `$event` object.
 If the target event is a native DOM element event, then `$event` is a
 [DOM event object](https://developer.mozilla.org/en-US/docs/Web/Events),
 with properties such as `target` and `target.value`.
 
 Consider this example:
+<!-- 
+```html
+<input [value]="currentItem.name"
+       (input)="currentItem.name=$event.target.value" >
```
Comment (brandonroberts) [inline aio/content/guide/template-syntax.md]: Remove the extra space between the `" >`

File: aio/content/examples/event-binding/src/app/app.component.html [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+</div>
+
+<div class="group">
+  <h3>$event and event handling statements</h3>
+  <h4>Result: {{currentItem.name}}</h4>
+
+  <!-- #docregion event-binding-3-->
+  <input [value]="currentItem.name"
+         (input)="currentItem.name=$event.target.value" >
+  without NgModel
```
Comment (brandonroberts) [inline aio/content/examples/event-binding/src/app/app.component.html]: Should "without NgModel" be included in the snippet?

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: aio/content/guide/template-syntax.md (hunk 1 above)
Comment (kapunahelewong) [reply aio/content/guide/template-syntax.md]: Great catch, fixed!

File: aio/content/guide/template-syntax.md (hunk 2 above)
Comment (kapunahelewong) [reply aio/content/guide/template-syntax.md]: Fixed.

File: aio/content/guide/template-syntax.md (hunk 3 above)
Comment (kapunahelewong) [reply aio/content/guide/template-syntax.md]: Deleted this whole section as the earlier docregion fix included the correct snippet.

File: aio/content/guide/template-syntax.md [hunk 5]
```diff
@@ -978,71 +965,78 @@ in response to the event, such as storing a value from the HTML control
 into a model.
 
 The binding conveys information about the event, including data values, through
-an **event object named `$event`**.
+an event object named `$event`.
```
[the remark this answers is not in the document]
Comment (kapunahelewong) [reply aio/content/guide/template-syntax.md]: Fixed. Thanks!

**Review by brandonroberts — APPROVED** [review]

**Review by brandonroberts — APPROVED** [review]

**Comment by brandonroberts** [comment]
Caretaker note: This is a docs-only change and I am using a global approval.

**Pull request closed without merging**